packages/cli/src/node-types.ts

```diff
@@ -66,9 +66,22 @@ export class NodeTypes implements INodeTypes {
 
 		if (type in knownNodes) {
 			const { className, sourcePath } = knownNodes[type];
-			const loaded: INodeType = loadClassInIsolation(sourcePath, className);
-			NodeHelpers.applySpecialNodeParameters(loaded);
+			const loaded: INodeType | IVersionedNodeType = loadClassInIsolation(sourcePath, className);
+			if (NodeHelpers.isINodeType(loaded)) NodeHelpers.applySpecialNodeParameters(loaded);
+
 			loadedNodes[type] = { sourcePath, type: loaded };
+
+			if (loaded.description.usableAsTool) {
+				const wrappedNode = NodeHelpers.convertNodeToAiTool(loaded);
```

Comment:
Shouldn't we wrap the node only if it's not `IVersionedNodeType`?

Reply:
Not really. This should work per version, and I've updated the code. Now, we do the wrapping after the versioned node is retrieved so that we can enable this only for a specific version